File: src/main/swift/Sources/swift/fr/jhelp/kotlinLightSamples/Element.swift

public class Element<T> {

    private let element: T
    private var next: Element<T>?

    public init(_ element: T){
        self.element = element
        self.next = nil
    }

    public func getElement() -> T
    {
        return self.element
    }

    public func getNext() -> Element<T>?
    {
        return self.next
    }

    public func setNext(_ element: T)
    {
        self.next = Element(element)
    }
}

File: src/main/kotlin/fr/jhelp/kotlinLightSamples/Element.kt
package fr.jhelp.kotlinLightSamples

class Element<T>
{
    private val element: T
    private var next: Element<T>?

    constructor(element: T)
    {
        this.element = element
        this.next = null
    }

    fun getElement(): T
    {
        return this.element
    }

    fun getNext(): Element<T>?
    {
        return this.next
    }

    fun setNext(element: T)
    {
        this.next = Element(element)
    }
}